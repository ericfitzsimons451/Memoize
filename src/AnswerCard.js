import React from 'react';
import './AnswerCard.scss';

function AnswerCard(props) {
  return(
    <div className="answer-card" id={props.questionInfo.id}>
      <h3 className="answer-card--question">{props.questionInfo.question}</h3>
      <h3 className="answer-card--correct-answer">Correct: {props.questionInfo.correctAnswer}</h3>
      <a href={props.questionInfo.link} className="answer-card--link" target="_blank" rel="noopener noreferrer">More Info</a>
      <button onClick={props.putInLocalStorage}>Store Answer For Later</button>
    </div>
  )
}

export default AnswerCard;