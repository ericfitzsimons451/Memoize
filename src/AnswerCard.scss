.answer-card {
  width: 600px;
  height: 650px;
  border: 1px solid black;
  background-color: #778078;
  background-image: url("https://www.transparenttextures.com/patterns/asfalt-dark.png");
  display: flex;
  flex-direction: column;
  margin: 5px;
  border-radius: 15px;
  justify-content: space-around;
}

.answer-card--question {
  font-size: 2.3rem;
  padding: 7px;
  margin: 5px;
}

.answer-card--correct-answer {
  font-size: 1.7rem;
  margin: 0px 20px 0px 20px;
  padding: 7px;
  background-color: white;
  border-radius: 15px;
}

.answer-card--link {
  font-size: 2.3rem;
  margin: 15px;
}

.answer-card--save-button, 
.answer-card--return-button {
  height: 80px;
  font-size: 1.7rem;
  margin: 0px 20px 0px 20px;
  padding: 3px;
  background-color: white;
  border-radius: 15px;
  &:hover {
    cursor: pointer;
  }
}

.answer-card--text {
  font-size: 1.8rem;
  padding: 7px;
  margin: 5px;
}
