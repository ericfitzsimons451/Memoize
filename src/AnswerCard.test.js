import React from 'react';
import AnswerCard from './AnswerCard.js';
import { shallow } from 'enzyme';

const currentAnswerToDisplay = { "id": 1,
    "categories":"Context",
    "question": "What is the default value of the THIS keyword?",
    "answer1": "The DOM Tree",
    "answer2": "The Global Window Object", 
    "answer3": "The Call Stack",
    "correctAnswer": "The Global Window Object",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this"
  }

const mockPutInLocalStorage = jest.fn();
const mockReturnToDisplayAll = jest.fn();

describe('AnswerCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AnswerCard questionInfo={currentAnswerToDisplay}
                  putInLocalStorage={mockPutInLocalStorage}
                  returnToDisplayAll={mockReturnToDisplayAll} />
      )
  })

  it('should match the snapshot with all data passed down', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should register a click when Save and Remove from Question Bank is clicked', () => {
    wrapper.find('.answer-card--save-button').simulate('click')
    expect(mockPutInLocalStorage).toBeCalled()
  })

  it('should register a click when Return Without Save is clicked', () => {
    wrapper.find('.answer-card--return-button').simulate('click')
    expect(mockReturnToDisplayAll).toBeCalled();
  })
})