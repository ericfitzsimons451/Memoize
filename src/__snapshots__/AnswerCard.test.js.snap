// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerCard should match the snapshot with all data passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnswerCard
    putInLocalStorage={[MockFunction]}
    questionInfo={
      Object {
        "answer1": "The DOM Tree",
        "answer2": "The Global Window Object",
        "answer3": "The Call Stack",
        "categories": "Context",
        "correctAnswer": "The Global Window Object",
        "id": 1,
        "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this",
        "question": "What is the default value of the THIS keyword?",
      }
    }
    returnToDisplayAll={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="answer-card--correct"
        >
          CORRECT!
        </h2>,
        <h3
          className="answer-card--question"
        >
          1
          ) 
          What is the default value of the THIS keyword?
        </h3>,
        <h3
          className="answer-card--correct-answer"
        >
          Correct Answer: 
          The Global Window Object
        </h3>,
        <h3
          className="answer-card--text"
        >
          Are you confident about this information?
        </h3>,
        <button
          className="answer-card--save-button"
          onClick={[MockFunction]}
        >
          Save and Remove from Question Bank
        </button>,
        <h3
          className="answer-card--text"
        >
          Not so confident?  Go back to the question bank.  You can try again later!
        </h3>,
        <button
          className="answer-card--return-button"
          onClick={[MockFunction]}
        >
          Return without save
        </button>,
        <a
          className="answer-card--link"
          href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this"
          rel="noopener noreferrer"
          target="_blank"
        >
          Want More Info?
        </a>,
      ],
      "className": "answer-card",
      "id": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "CORRECT!",
          "className": "answer-card--correct",
        },
        "ref": null,
        "rendered": "CORRECT!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            ") ",
            "What is the default value of the THIS keyword?",
          ],
          "className": "answer-card--question",
        },
        "ref": null,
        "rendered": Array [
          1,
          ") ",
          "What is the default value of the THIS keyword?",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Correct Answer: ",
            "The Global Window Object",
          ],
          "className": "answer-card--correct-answer",
        },
        "ref": null,
        "rendered": Array [
          "Correct Answer: ",
          "The Global Window Object",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Are you confident about this information?",
          "className": "answer-card--text",
        },
        "ref": null,
        "rendered": "Are you confident about this information?",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save and Remove from Question Bank",
          "className": "answer-card--save-button",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Save and Remove from Question Bank",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not so confident?  Go back to the question bank.  You can try again later!",
          "className": "answer-card--text",
        },
        "ref": null,
        "rendered": "Not so confident?  Go back to the question bank.  You can try again later!",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return without save",
          "className": "answer-card--return-button",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Return without save",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Want More Info?",
          "className": "answer-card--link",
          "href": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Want More Info?",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="answer-card--correct"
          >
            CORRECT!
          </h2>,
          <h3
            className="answer-card--question"
          >
            1
            ) 
            What is the default value of the THIS keyword?
          </h3>,
          <h3
            className="answer-card--correct-answer"
          >
            Correct Answer: 
            The Global Window Object
          </h3>,
          <h3
            className="answer-card--text"
          >
            Are you confident about this information?
          </h3>,
          <button
            className="answer-card--save-button"
            onClick={[MockFunction]}
          >
            Save and Remove from Question Bank
          </button>,
          <h3
            className="answer-card--text"
          >
            Not so confident?  Go back to the question bank.  You can try again later!
          </h3>,
          <button
            className="answer-card--return-button"
            onClick={[MockFunction]}
          >
            Return without save
          </button>,
          <a
            className="answer-card--link"
            href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this"
            rel="noopener noreferrer"
            target="_blank"
          >
            Want More Info?
          </a>,
        ],
        "className": "answer-card",
        "id": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CORRECT!",
            "className": "answer-card--correct",
          },
          "ref": null,
          "rendered": "CORRECT!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              ") ",
              "What is the default value of the THIS keyword?",
            ],
            "className": "answer-card--question",
          },
          "ref": null,
          "rendered": Array [
            1,
            ") ",
            "What is the default value of the THIS keyword?",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Correct Answer: ",
              "The Global Window Object",
            ],
            "className": "answer-card--correct-answer",
          },
          "ref": null,
          "rendered": Array [
            "Correct Answer: ",
            "The Global Window Object",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Are you confident about this information?",
            "className": "answer-card--text",
          },
          "ref": null,
          "rendered": "Are you confident about this information?",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save and Remove from Question Bank",
            "className": "answer-card--save-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Save and Remove from Question Bank",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not so confident?  Go back to the question bank.  You can try again later!",
            "className": "answer-card--text",
          },
          "ref": null,
          "rendered": "Not so confident?  Go back to the question bank.  You can try again later!",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return without save",
            "className": "answer-card--return-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Return without save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Want More Info?",
            "className": "answer-card--link",
            "href": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Want More Info?",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
